packages/components/src/shared/src/useIsInitialRender.ts

```diff
@@ -8,5 +8,5 @@ export function useIsInitialRender() {
         setIsInitial(false);
     }, [setIsInitial]);
 
-    return isInitialRef.current;
+    return isInitialRef;
```

Comment:
These kind of changes are dangerous in the long run.

By that, I mean, if you had to design this hook first, you would probably have returned a bool instead of a ref.

Now you have to solve a specific issue somewhere in the code base so you change the API to return a ref instead.

Maybe the problem is somewhere else or there is another solution which doesn't involve reworking the API of a core hook?

Reply:
I fixed it another way, by wrapping the useInitialRender in a committedRef.